cfme/cloud/keypairs.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+import cfme.fixtures.pytest_selenium as sel
+from cfme.common import SummaryMixin, Taggable
+from cfme.web_ui import (Quadicon, flash, Form, Input, form_buttons, fill, summary_title,
+    breadcrumbs, AngularSelect, Table)
+from functools import partial
+from cfme.web_ui import toolbar as tb
+from utils.wait import wait_for
+
+
+cfg_btn = partial(tb.select, "Configuration")
+keypair_form = Form(
+    fields=[
+        ('name', Input("name")),
+        ('public_key', Input("public_key")),
+        ('provider', AngularSelect('ems_id')),
+        ('save_button', form_buttons.add)
+    ])
+
+records_table = Table("//div[@id='list_grid']//table")
+
+
+class KeyPair(Taggable, SummaryMixin):
+    """ Model of KeyPairs
+
+    Args:
+        name: Name of Keypairs.
+    """
+
+    def __init__(self, name=None):
+        self.name = name
+
+    def _on_detail_page(self):
+        title = '{} (Summary)'.format(self.name)
+        return breadcrumbs() == ['Key Pairs', title] and summary_title() == title
+
+    def load_details(self, refresh=False):
+        if not self._on_detail_page():
+            self.navigate(detail=True)
+        elif refresh:
+            tb.refresh()
+
+    def delete(self):
+        sel.force_navigate('clouds_key_pairs', context={'keypairs': self.name})
+        row = records_table.find_row_by_cells({'Name': self.name})
+        sel.check(sel.element(".//input[@type='checkbox']", root=row[0]))
```

Comment:
Shouldn't this be rather a CheckboxTable, when interacting with this?

And when deleting, we usually load_details and then tb.select, we don't delete from the table view by default.